addons/op-badge/userscript.js

```diff
@@ -0,0 +1,16 @@
+export default async function({ addon, console, msg }) {
+    const posts = document.querySelectorAll(".blockpost");
+
+    const html = new DOMParser().parseFromString(await fetch(`https://scratch.mit.edu/discuss/topic/${window.location.href.split("/")[5]}/`).then(res => res.text()), "text/html");
```

Comment:
If this is already running on the first page, we can set `html` to `document` and avoid the DOMParser call and the fetch.

Reply:
Self TODO: it would also be better if the fetch call was unauthenticated to let Scratch hit the cache. Haven't checked whether current code sends auth cookies or not.